upup/pkg/fi/nodeup/nodetasks/service.go

```diff
@@ -75,9 +75,19 @@ func (p *Service) GetDependencies(tasks map[string]fi.Task) []fi.Task {
 			deps = append(deps, v)
 		case *Service, *LoadImageTask:
 			// ignore
+
 		default:
-			klog.Warningf("Unhandled type %T in Service::GetDependencies: %v", v, v)
-			deps = append(deps, v)
+			switch fmt.Sprintf("%T", v) {
+			case "*model.KubeletBootstrapKubeconfigTask":
```

Comment:
Why not add this type to the outer switch? And why test the type by converting to string?

Reply:
I'll add a comment, but it's because of go packages not allowing circular references..